src/components/ec-checkbox/ec-checkbox.vue

```diff
@@ -80,49 +80,46 @@
   </div>
 </template>
 
-<script setup>
+<script setup lang="ts">
 defineOptions({
   inheritAttrs: false,
 });
 
 import {
   computed,
+  type InputHTMLAttributes,
   onMounted,
   ref,
+  type StyleValue,
   useSlots,
   watch,
 } from 'vue';
 
 import { getUid } from '../../utils/uid';
 import EcIcon from '../ec-icon';
+import { IconName } from '../ec-icon/types';
+import { CheckboxEvent, type CheckboxEvents } from './types';
+
+interface CheckboxProps {
+  modelValue?: boolean,
```

Comment:
shouldn't modelValue be required?

Reply:
For DX we want the user of the component to have optional control of the component, similar to the ec-input-field. For example we may want to show a disabled checked checkbox with no ability for the user to interact with, so forcing a modelValue would be unnecessary binding.